Mathlib/Condensed/Equivalence.lean

```diff
@@ -0,0 +1,306 @@
+
+/-- The morphism from `presentation X` to `X`. -/
+noncomputable
+def presentation.π (X : Profinite) : Stonean.toProfinite.obj X.presentation ⟶ X :=
+  X.toCompHaus.projectivePresentation.f
+
+/-- The morphism from `presentation X` to `X` is an epimorphism. -/
+noncomputable
+instance presentation.epi_π (X : Profinite) : Epi (π X) := by
+  have := X.toCompHaus.projectivePresentation.epi
+  rw [CompHaus.epi_iff_surjective] at this
+  rw [Profinite.epi_iff_surjective]
+  exact this
+
+/--
+```
+               X
+               |
+              (f)
+               |
+               \/
+  Z ---(e)---> Y
+```
+If `Z` is extremally disconnected, X, Y are profinite, if `f : X ⟶ Y` is an epi and
+`e : Z ⟶ Y` is arbitrary, then `lift e f` is a fixed (but arbitrary) lift of `e` to a morphism
+`Z ⟶ X`. It exists because `Z` is a projective object in `CompHaus`.
+-/
+noncomputable
+def lift {X Y : Profinite} {Z : Stonean} (e : Stonean.toProfinite.obj Z ⟶ Y) (f : X ⟶ Y) [Epi f] :
+    Stonean.toProfinite.obj Z ⟶ X := CompHaus.lift e f
+
+@[simp, reassoc]
+lemma lift_lifts {X Y : Profinite} {Z : Stonean} (e : Stonean.toProfinite.obj Z ⟶ Y) (f : X ⟶ Y)
+    [Epi f] : lift e f ≫ f = e := CompHaus.lift_lifts _ _
+
+lemma projective_of_extrDisc {X : Profinite} (hX : ExtremallyDisconnected X) :
+    Projective X:= by
+  let X' : Stonean := ⟨X.toCompHaus⟩
+  show Projective (Stonean.toProfinite.obj X')
+  exact inferInstance
+
+end Profinite
+
+end ProjectiveProfinite
+
+namespace Condensed
+
+universe u w
+
+namespace StoneanProfinite
+
+open Profinite in
+lemma generate_singleton_mem_coherentTopology (B : Profinite) :
+    Sieve.generate (Presieve.singleton (presentation.π B)) ∈ coherentTopology Profinite B := by
+  apply Coverage.saturate.of
+  refine ⟨Unit, inferInstance, fun _ => (Stonean.toProfinite.obj B.presentation),
+    fun _ => (presentation.π B), ?_ , ?_⟩
+  · funext X f
+    ext
+    refine ⟨fun ⟨⟩ ↦ ⟨()⟩, ?_⟩
+    rintro ⟨⟩
+    simp only [Presieve.singleton_eq_iff_domain]
+  · apply ((effectiveEpiFamily_tfae
+      (fun (_ : Unit) => (Stonean.toProfinite.obj B.presentation))
+      (fun (_ : Unit) => (Profinite.presentation.π B))).out 0 2).mpr
+    intro b
+    exact ⟨(), (Profinite.epi_iff_surjective (presentation.π B)).mp (presentation.epi_π B) b⟩
+
+open Profinite in
+lemma coverDense : CoverDense (coherentTopology _) Stonean.toProfinite := by
+  constructor
+  intro B
+  convert generate_singleton_mem_coherentTopology B
+  ext Y f
+  refine ⟨fun ⟨⟨obj, lift, map, fact⟩⟩ ↦ ?_, fun ⟨Z, h, g, hypo1, hf⟩ ↦ ?_⟩
+  · obtain ⟨p, p_factors⟩ := Projective.factors map (Profinite.presentation.π B)
+    exact ⟨(Stonean.toProfinite.obj (presentation B)), ⟨lift ≫ p, ⟨(presentation.π B),
+      ⟨Presieve.singleton.mk, by rw [Category.assoc, p_factors, fact]⟩⟩⟩⟩
+  · cases hypo1
+    exact ⟨⟨presentation B, h, presentation.π B, hf⟩⟩
+
+theorem coverDense.inducedTopology_Sieve_iff_EffectiveEpiFamily (X : Stonean) (S : Sieve X) :
```

Comment:
If this really needed? I had the impression that we now had this for the coherent topology on any precoherent category.

Reply:
I think this lemma together with `coherentTopology.mem_sieves_iff_hasEffectiveEpiFamily` that you mentioned shows that the induced topology coincides with the coherent topology.